code/modules/recycling/conveyor2.dm

```diff
@@ -127,16 +128,29 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 /obj/machinery/conveyor/process()
 	if(stat & (BROKEN | NOPOWER))
 		return
-	if(!operating)
+	//If the conveyor is broken or already moving items
+	if(!operating || conveying)
 		return
 	use_power(6)
-	affecting = loc.contents - src		// moved items will be all in loc
+	//get the first 30 items in contents
+	var/i = 0
+	for(var/item in loc.contents)
+		i++
+		if(item == src)
+			continue
+		if(i >= MAX_CONVEYOR_ITEMS_MOVE)
+			break
+		affecting.Add(item)
+	conveying = TRUE
 	addtimer(CALLBACK(src, .proc/convey, affecting), 1)
 
 /obj/machinery/conveyor/proc/convey(list/affecting)
 	for(var/atom/movable/A in affecting)
-		if((A.loc == loc) && A.has_gravity())
```

Comment:
has gravity is already checked in conveyormove fyi

Reply:
Y'all better not be stealing credit from me, i want my fake internet thumbs up too

Reply:
fair enough buggy found that